### PR TITLE
Rewrite circuitbox configuration

### PR DESCRIPTION
**Pull request by matthewshafer**
Circuitbox supports custom configuration but it uses instance_eval to do the configuration. It also resets the configuration when reset is called on either Circuitbox or Circuitbox::CircuitBreaker.

These changes remove reset at both the Circuitbox and circuit breaker level, remove the class variables from Circuitbox, expose a configuration method that handles clearing Circuitbox's cache of known circuits, and add a new default_logger which is useful to share a custom logger between circuits.

### CODE REVIEW

File: lib/circuitbox.rb [hunk 1]
```diff
@@ -11,71 +10,29 @@
...
-    else
-      self.instance.circuits[service_name]
+      circuit.run { yield }
     end
-  end
 
-  def self.parameter_to_service_name(param)
-    uri = URI(param.to_s)
-    uri.host.present? ? uri.host : param.to_s
+    def parameter_to_service_name(param)
```
Comment (bmorton) [inline lib/circuitbox.rb]: Should this be private?

File: lib/circuitbox/circuit_breaker.rb [hunk 2]
```diff
@@ -25,14 +25,14 @@ class CircuitBreaker
     def initialize(service, options = {})
       @service = service.to_s
       @circuit_options = options
-      @circuit_store   = options.fetch(:cache) { Circuitbox.circuit_store }
+      @circuit_store   = options.fetch(:cache) { Circuitbox.default_circuit_store }
```
Comment (bmorton) [inline lib/circuitbox/circuit_breaker.rb]: Worth unifying these names at this point?  `cache` vs `circuit_store`?

Comment (matthewshafer) [reply lib/circuitbox/circuit_breaker.rb]: I'll change that in another pr

File: lib/circuitbox.rb (hunk 1 above)
Comment (matthewshafer) [reply lib/circuitbox.rb]: yeah, theres some tests that need to be fixed too. I find this method a bit interesting, to me it only makes sense to do something like this from the faraday/excon middleware. I wonder if it's worth getting rid of this (or moving use of it to another public method) to speed up accessing circuits through the circuit method.

**Pull request merged**